### DISCUSSION TITLE
Spatie Media Library Plugin MorphOne Support ( using ->one() )

### DISCUSSION THREAD

**Discussion by SanderCokart** [filamentphp/filament#6108] [Ideas]
So right now there is a new method for MorphMany relationship to turn it into MorphOne, example:
```php
 public function banner(): MorphOne
    {
        return $this->media()->where('collection_name', MediaCollectionEnum::ArticleBanners->name)->one();
    }
```
However this breaks `Tables\Columns\SpatieMediaLibraryImageColumn::class`
Example: 
```php
//Doesn't work
Tables\Columns\SpatieMediaLibraryImageColumn::make('banner')->collection(MediaCollectionEnum::ArticleBanners->name),
//Somehow works making it plural...
Tables\Columns\SpatieMediaLibraryImageColumn::make('banners')->collection(MediaCollectionEnum::ArticleBanners->name),
```

It would be nice if the plugin supported receiving a MorphOne using maybe an if statement with instanceof,
so we can continue to use singular names.

**Comment by o-tari** [comment]
I had the same issue, I was forced to use MorphMany
